subsystem/core/src/main/java/org/wildfly/extension/camel/service/CamelEndpointDeployerService.java

```diff
@@ -0,0 +1,497 @@
+        protected void service(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+            endpointHttpHandler.service(getServletContext(), req, res);
+        }
+
+        public void setEndpointHttpHandler(EndpointHttpHandler endpointHttpHandler) {
+            this.endpointHttpHandler = endpointHttpHandler;
+        }
+
+    }
+
+    static class UrlPattern {
+
+        private final String source;
+
+        public UrlPattern(String source) {
+            super();
+            this.source = source;
+        }
+
+        public String relativize(String path) {
+            final StringTokenizer patternTokenizer = new StringTokenizer(source, "/", true);
+            final StringTokenizer pathTokenizer = new StringTokenizer(path, "/", true);
+            /* skip the common segments */
+            while (pathTokenizer.hasMoreTokens()) {
+                if (!patternTokenizer.hasMoreTokens()) {
+                    return null;
+                }
+                final String patternToken = patternTokenizer.nextToken();
+                final String pathToken = pathTokenizer.nextToken();
+                final boolean isSlash = patternToken.equals("/");
+                if (isSlash != pathToken.equals("/")) {
+                    return null;
+                } else if (!isSlash) {
+                    if ("*".equals(pathToken)) {
+                        continue;
+                    } else {
+                        final StringTokenizer segmentTokenizer = new StringTokenizer(patternToken, "*", true);
+                        final StringBuilder segmentPatternBuilder = new StringBuilder(patternToken.length());
+                        while (segmentTokenizer.hasMoreTokens()) {
+                            final String segmentToken = segmentTokenizer.nextToken();
+                            if ("*".equals(segmentToken)) {
+                                segmentPatternBuilder.append(".*");
+                            } else {
+                                segmentPatternBuilder.append(Pattern.quote(segmentToken));
+                            }
+                        }
+                        final Pattern pat = Pattern.compile(segmentPatternBuilder.toString());
+                        if (!pat.matcher(pathToken).matches()) {
+                            return null;
+                        }
+                    }
+                }
+            }
+            final StringBuilder result = new StringBuilder();
+            while (patternTokenizer.hasMoreTokens()) {
+                result.append(patternTokenizer.nextToken());
+            }
+            if (result.length() == 0) {
+                return "";
+            } else if (result.charAt(0) != '/') {
+                return "/" + result.toString();
+            } else {
+                return result.toString();
+            }
+        }
+
+    }
+
+    private static final String CATCH_ALL_ENDPOINT_URI_PREFIX = "///*";
```

Comment:
Can we keep constants at the top of the class please.

Reply:
Eclipse puts inner classes before static fields by default. I am ready to adhere to any other ordering if you please tell me the complete definition?

Reply:
Ok, that's a weird convention. I hate seeing fields and static fields spread at random throughout the class. They're better placed all together at the top IMO.

Maybe we should consider introducing checkstyle or add Sonar integration to PRs.

Reply:
Strange enough, that neither checkstyle's DeclarationOrderCheck [1] nor Sun's Code Conventions for the Java Programming Language [2] say where the inner classes should be placed.

[1] http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.html 

[2] http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141855.html#1852